application/templates/deployment.yaml

```diff
@@ -291,9 +291,11 @@ spec:
         {{- end }}
         {{- end }}
       {{- end }}
+      {{- if .Values.rbac.serviceAccount.enabled }}
       {{- if .Values.rbac.serviceAccount.name }}
       serviceAccountName: {{ .Values.rbac.serviceAccount.name }}
-        {{- else }}
-      serviceAccountName: {{ template "application.name" . }}
+      {{- else }}
+      serviceAccountName: {{ template "application.name" $ }}
+      {{- end }}
       {{- end }}
```

Comment:
I think it's worth adding a chart unit test to check that the `serviceAccountName` field is empty. We have three possible branches here: populated and proxied from values, populated via the fallback (`template "application.name" $`), and empty.

At the very least we should test that it's empty when disabled:

```yaml
- it: yields empty service account name when disabled
  set:
    rbac.serviceAccount.enabled: false
  asserts:
    - isEmpty:
        path: spec.template.spec.serviceAccountName
```

Reply:
I added tests to cover the three branches. Let me know if it fits you :)